### PR TITLE
fix kubeContext does not globally defaulting its value

### PR DESCRIPTION
**Pull request by katsew**
related issue: https://github.com/roboll/helmfile/issues/1810#issuecomment-833268566

### CODE REVIEW

**Comment by katsew** [comment]
@mumoshu 

Hi, can you fix this error?

```
Error: looks like "https://repo.chartcenter.io" is not a valid chart repository or cannot be reached: failed to fetch https://repo.chartcenter.io/index.yaml : 405 Not Allowed
```

I think this is not a problem of a test itself but its environment.
I run `make test` locally and all tests have passed.

**Comment by mumoshu** [comment]
@katsew CI has been fixed via https://github.com/roboll/helmfile/pull/1833 👍

File: pkg/state/state_exec_tmpl.go [hunk 1]
```diff
@@ -94,6 +90,9 @@ func (st *HelmState) ExecuteTemplates() (*HelmState, error) {
 	vals := st.Values()
 
 	for i, rt := range st.Releases {
+		if rt.KubeContext == "" {
+			rt.KubeContext = r.HelmDefaults.KubeContext
+		}
```
Comment (mumoshu) [inline pkg/state/state_exec_tmpl.go:95]: Ah, nice. This looks like the right place to do defaulting. I might rename `ExecuteTemplates` later to reflect that... perhaps `ApplyDefaultsAndReleaseTemplates` 🤣

**Review by mumoshu — APPROVED** [review]
LGTM. Thanks a lot for your continued effort @katsew ☺️

**Pull request merged**